dashboard/src/reducers/charts.ts

```diff
@@ -88,8 +88,6 @@ const chartsReducer = (
         isFetching: false,
         selected: chartsSelectedReducer(state.selected, action),
       };
-    case getType(actions.namespace.setNamespaceState):
-      return { ...initialState };
```

Comment:
This was deleting the chart state when changing the namespace. I believe this was to re-fetch chart information when checking different applications but it was not necessary. 

In fact, this was causing an issue: Since charts don't depend on the context namespace (they depend on the chart namespace), after cleaning up the state, charts were not being re-fetched, since functions that fetch them don't depend on the namespace.

Reply:
It may have been before certain data was stored per namespace in the state. I'd think we'd certainly want to ensure that any data which is not stored per namespace is either cleared or updated to be stored per namespace, but yeah, resetting to the initial state may no longer be necessary if things are stored per namespace.

Reply:
I am not sure if I'm following your reasoning but no data is stored per namespace (unless the namespace info itself). The way it's structured now, the app info or the chart info is stored independently and it may require to be re-fetched when the namespace change.

In this case, the chart state doesn't depend on the user namespace but on the chart namespace. This will be the same for namespaced repositories but no for global repositories That's why it should not be cleaned up when changing namespaces.

Reply:
My reasoning was: if we only store the current chart in the state (rather than charts per namespace), then when the user switches namespaces we'd want to avoid displaying a chart which may not be relevant in the new namespace (the issue), so we'd need to be sure we clear the state (which is what I think this code intended, but as you say, didn't account for the difference between the user-selected namespace and the chart namespace, but you've fixed that by moving it).

If we stored charts per (chart) namespace, we wouldn't need to do this, I don't think? (though there may be other issues doing that). Anyway, I'm fine with the way you've chosen.

Reply:
> If we stored charts per (chart) namespace, we wouldn't need to do this, I don't think? (though there may be other issues doing that). Anyway, I'm fine with the way you've chosen.

yes, we could restructure the store to be aware of the chart namespace (and in general, that's something we could do for improving performance and caching stuff) but it would require a slighly big refactor that I am not sure would pay off for the benefit it gives.